src/main/java/com/chromediopside/service/ErrorService.java

```diff
@@ -34,12 +34,12 @@ String missingValues(LoginForm loginForm) {
 
     }
     Collections.sort(missingFields);
-    return String.join(", ", missingFields) + "!";
+    return String.join(", ", missingFields);
   }
 
   public ErrorResponse fieldErrors(LoginForm loginForm) {
     errorResponse.setStatus("error");
-    errorResponse.setMessage(MISSING_PARAMS_MESSAGE + missingValues(loginForm));
+    errorResponse.setMessage(MISSING_PARAMS_MESSAGE + missingValues(loginForm) + "!");
```

Comment:
but it's not a problem if you leave this, because we are concatenating just a few Strings here.